src/lib.rs

```diff
@@ -183,3 +183,17 @@ fn clockpoly(
     polygon
 }
 
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn internal() {
+        let gj = clockboard();
+        // check that the number of features is correct, in R
+        // gj = sf::read_sf("zones.geojson")
+        // nrow(gj)
+        // 60
+        assert_eq!(60, gj.length());
```

Comment:
Using your IDE's autocomplete or searching through the API docs is the way to figure out the possible methods you can call on something. `clockboard` returns the [GeoJson](https://docs.rs/geojson/0.22.2/geojson/enum.GeoJson.html) type. The docs say that it's an `enum` with 3 cases. https://doc.rust-lang.org/rust-by-example/custom_types/enum.html explains enums pretty well -- basically the `GeoJson` could be any of these 3 cases. Here you know that it's `FeatureCollection`, because of how `clockboard` assembles the final result. So you can unpack that case of the enum by doing:

```
if let GeoJson::FeatureCollection(fc) = gj {
  // Now the fc variable is https://docs.rs/geojson/0.22.2/geojson/struct.FeatureCollection.html
  // Looking at those docs, it has 'pub features: Vec<Features>'
  // So...
  assert_eq!(60, fc.features.len());
}
```

But if for some reason you're wrong about the assumption that the geojson object is a feature collection, you want the unit test to tell you; as it's written now, it would just silently pass, because `if let` just looks for one case of an enum. You could just tack on an `} else { panic!("not a feature collection"); }` or something.

So all in all:

```
    #[test]       
    fn internal() {
        let args: Vec<String> = Vec::new();
        let gj = clockboard(Point::new(0.0, 0.0), Params::from_iter(args));
        if let GeoJson::FeatureCollection(fc) = gj {
            assert_eq!(60, fc.features.len());
        } else {
            panic!("not a feature collection");
        }                                                                                              
    }
```

Reply:
Many thanks for the feedback. I spent some time looking for precisely that `features.len()` method, clearly need to get better and parsing Rust crate docs!

Will update and retry :construction: